### DISCUSSION TITLE
Error in rule init pre assembly processing metatranscriptomes

### DISCUSSION THREAD

**Discussion by Magdalena-PP** [metagenome-atlas/atlas#619] [General]
Hi @SilasK ,

My name is Magda, I am working with @Sofie8 on metatranscriptomics data from soil. I got an error in rule init pre assembly processing .
[rule init_pre_assembly_process.txt](https://github.com/metagenome-atlas/atlas/files/10979955/rule.init_pre_assembly_process.txt)

Could you please help me with this? 

I use Atlas version: 2.14.2 Snakemake 7.18.2.

Best regards,
Magda

**Comment by SilasK** [comment]
It seems your OS doesn't allow you to make the symlink `/home/magdalenapp/NASSerwMPP/metatrans_out_new/S15/sequence_quality_control/S15_QC_R1.fastq.gz' -> 'S15/assembly/reads/QC_R1.fastq.gz`


There are some filesystems that may not permit symlinks, but Sofie never had such a problem. 

Could it be that you took over the project from Sofie and therefore you don't have write permissions for some files/folders?


could you try to make 
```
cd <atlas dir>
mkdir -p S15/assembly/reads/
ln -s /home/magdalenapp/NASSerwMPP/metatrans_out_new/S15/sequence_quality_control/S15_QC_R1.fastq.gz S15/assembly/reads/QC_R1.fastq.gz

# and maybe 
ln -s S15/sequence_quality_control/S15_QC_R1.fastq.gz S15/assembly/reads/QC_R1.fastq.gz

```

**Comment by SilasK** [comment]
@Magdalena-PP You also want to try to assemble metatranscriptome only?

In this case I suggest you to install atlas from Github (see the docs) and use the branch `transcriptome` .


https://github.com/metagenome-atlas/atlas/pull/615

**Reply by SilasK** [reply]
Have you seen this?

**Comment by Magdalena-PP** [comment]
Hi 😊

 

Sofie is analyzing my samples on her computer, and at the same time, I am trying to do it on my computer. 

 

I’ve tried the commands you proposed, and it didn’t work ☹ 

 

(snakemake) ***@***.***:~$ ln -s S15/sequence_quality_control/S15_QC_R1.fastq.gz S15/assembly/reads/QC_R1.fastq.gz

ln: failed to create symbolic link 'S15/assembly/reads/QC_R1.fastq.gz': No such file or directory

 

Do you maybe know if I can get permission for symlink on my OS?

 

Greetings,

Magda

 

--

Dr Magdalena Pacwa-Płociniczak

Assistant Professor



University of Silesia in Katowice
Faculty of Natural Sciences | Institute of Biology, Biotechnology and Environmental Protection
Jagiellonska 28, 40-032 Katowice POLAND
tel. +48322009442 | mob. +48607335458
e-mail:  ***@***.***> ***@***.***

 <http://www.us.edu.pl/> www.us.edu.pl

Please consider the environment before printing this email

**Comment by Magdalena-PP** [comment]
Dear Silas,

 

Thank you for being so helpful. Now I know why my OS didn’t allow me to make the symlink – the directory with the results was a shared folder.

 

BR,

Magda

--

Dr Magdalena Pacwa-Płociniczak

Assistant Professor



University of Silesia in Katowice
Faculty of Natural Sciences | Institute of Biology, Biotechnology and Environmental Protection
Jagiellonska 28, 40-032 Katowice POLAND
tel. +48322009442 | mob. +48607335458
e-mail:  ***@***.***> ***@***.***

 <http://www.us.edu.pl/> www.us.edu.pl

Please consider the environment before printing this email

**Reply by SilasK** [reply]
Thank you for writing back.

**Comment by SilasK** [comment]
Feel free to ask again if you encounter problems.

**Comment by Magdalena-PP** [comment]
Dear Silas,

 

Finally, I was able to finish my metatranscriptomic analysis, but the results I obtained were weird. I am sending it as an attachment. Do you maybe know the reason?

 

Best regards,

Magda

 

 

 

--

Dr Magdalena Pacwa-Płociniczak

Assistant Professor



University of Silesia in Katowice
Faculty of Natural Sciences | Institute of Biology, Biotechnology and Environmental Protection
Jagiellonska 28, 40-032 Katowice POLAND
tel. +48322009442 | mob. +48607335458
e-mail:  ***@***.***> ***@***.***

 <http://www.us.edu.pl/> www.us.edu.pl

Please consider the environment before printing this email

**Reply by SilasK** [reply]
You probably simply responded by mail. The attachment is not correctly shown on the GitHub page. https://github.com/metagenome-atlas/atlas/discussions/619

You can upload a zip archive here or otherwise you can find my e mail on my personal webpage. 

You are reffering to the genecatalog results are you? 

Can you also send me the stats and repors folder as zip?

**Reply by Magdalena-PP** [reply]
I am sending zip folder with genecatalog results and logs folder.
[Magda.zip](https://github.com/metagenome-atlas/atlas/files/11108930/Magda.zip)

**Reply by Magdalena-PP** [reply]
Hi @SilasK , did you maybe find the reason for my problem with the results from genecatalog?

**Reply by SilasK** [reply]
I have looked at your results. I don't know what you think is weird? Do you mean the parquet files? 

You can open them easily with 

```{python}
import pandas as pd
counts_file= "Genecatalog/counts/Nmapped_reads.parquet"

counts = pd.read_parquet(counts_file).set_index("GeneNr")

```


In r you have the `arrow::read_parquet` function

**Reply by SilasK** [reply]
From the Nmapped reads I calculated that you have this numbers of reads mapped:

S13    13031963
S14    11319618
S15    25824461
S16    10379891
S17    10440977
S18    10379891
S46    21432934
S47    24582128
S48    22212425
S49    12260911
S50    12405066
S51    18037556
S52    19974472
S53    13367263
S54    10709324

Could you check if this corresponds to the stats/readcounts.tsv?

**Reply by SilasK** [reply]
For your information, in the latest version of atlas I swicht to a hdf format for the coutns in the genecatalog, in order to better store genegatalogs for very large datasets. But for you the current is fine.

**Comment by SilasK** [comment]
Did you do an annotation, e.g. eggNOG?

**Comment by Magdalena-PP** [comment]
Yes, you are right; It was my mistake – because I am just learning bioinformatics – I’ve tried to open parquet files with excel… Now everything is OK. Thanks for your help 😊.

 

--

Dr Magdalena Pacwa-Płociniczak

Assistant Professor



University of Silesia in Katowice
Faculty of Natural Sciences | Institute of Biology, Biotechnology and Environmental Protection
Jagiellonska 28, 40-032 Katowice POLAND
tel. +48322009442 | mob. +48607335458
e-mail:  ***@***.***> ***@***.***

 <http://www.us.edu.pl/> www.us.edu.pl

Please consider the environment before printing this email
